docs/core-features/table_lookups_basics.md

```diff
@@ -0,0 +1,186 @@
+# Table lookup basics
+
+In TFHE, there exists mainly two operations: the linear operations (additions, subtractions, multiplications by integers) and the rest. And the rest is done with table lookups (TLUs), which means that a lot of things are done with TLU. In this document, we explain briefly, from a user point of view, how TLU can be used. In [the poweruser documentation](../dev/compilation/compiler_internals.md), we enter a bit more into the details.
+
+## Direct TLU
+
+Direct TLU stands for instructions of the form `y = T[i]`, for some table `T` and some index `i`. One can use the `fhe.LookupTable` to define the table values, and then use it on scalars or tensors.
+
+```python
+from concrete import fhe
+
+table = fhe.LookupTable([2, -1, 3, 0])
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+inputset = range(4)
+circuit = f.compile(inputset)
+
+assert circuit.encrypt_run_decrypt(0) == table[0] == 2
+assert circuit.encrypt_run_decrypt(1) == table[1] == -1
+assert circuit.encrypt_run_decrypt(2) == table[2] == 3
+assert circuit.encrypt_run_decrypt(3) == table[3] == 0
+```
+
+```python
+from concrete import fhe
+import numpy as np
+
+table = fhe.LookupTable([2, -1, 3, 0])
+
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+
+inputset = [np.random.randint(0, 4, size=(2, 3)) for _ in range(10)]
+circuit = f.compile(inputset)
+
+sample = [
+    [0, 1, 3],
+    [2, 3, 1],
+]
+expected_output = [
+    [2, -1, 0],
+    [3, 0, -1],
+]
+actual_output = circuit.encrypt_run_decrypt(np.array(sample))
+
+assert np.array_equal(actual_output, expected_output)
+```
+
+`LookupTable` mimics array indexing in Python, which means if the lookup variable is negative, the table is looked up from the back.
+
+## Indirect TLU
+
+Indirect TLU stands for instructions of the form `y = T[j][i]`, for some set of tables `T`, some table-index `j` and some index `i`. One can use the `fhe.LookupTable` to define the table values, and then use it on scalars or tensors.
+
+```python
+from concrete import fhe
+import numpy as np
+
+squared = fhe.LookupTable([i ** 2 for i in range(4)])
+cubed = fhe.LookupTable([i ** 3 for i in range(4)])
+
+table = fhe.LookupTable([
+    [squared, cubed],
+    [squared, cubed],
+    [squared, cubed],
+])
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+inputset = [np.random.randint(0, 4, size=(3, 2)) for _ in range(10)]
+circuit = f.compile(inputset)
+
+sample = [
+    [0, 1],
+    [2, 3],
+    [3, 0],
+]
+expected_output = [
+    [0, 1],
+    [4, 27],
+    [9, 0]
+]
+actual_output = circuit.encrypt_run_decrypt(np.array(sample))
+
+assert np.array_equal(actual_output, expected_output)
+```
+
+### TLU on the most significant bits
```

Comment:
Right before this, I'd describe `## Transparent TLU`, with examples like `x ** 2` and `fhe.univariate`. You could say we generate the tables appropriately, instead of you creating it.

You would be able to use those from now on as well.

Reply:
I would also promote the univariate and multivariate options, since they are nice and easy to understand. But then, do we agree we would need to remove them from the extensions.md? I try to avoid the things to be explained several times in several places.

Reply:
Maybe we could move the univariate and multivariate section to either table_lookups_basics.md or non_linear_operations.md part. On the bad side, it breaks the extension.md unity. On the good side, it promotes something which is easy to use and to understand

What do you think, @umut-sahin @BourgerieQuentin ?

Reply:
I think we can leave them on extensions and describe `x ** 2` as an indirect TLU example. And then mention (uni/multi)variate with a link to extensions.md as an info bubble.

We'd keep extension docs as is, show transparent tlus, and mention powerful univariate and multivariate at the same time!

Reply:
Adding "Transparent TLU", you'll tell me if you like it, @umut-sahin